client/pool.go

```diff
@@ -224,10 +231,14 @@ func (pool *Pool) putConnectionUnsafe(connection Connection) {
 }
 
 func (pool *Pool) newConnectionProducer() {
```

Comment:
I found another problem, please help me check it.
in line 271, `pool.readyConnection <- connection` may block the for loop. So if `Close` tries to drain `pool.readyConnection` first and then `newConnectionProducer` sends at channel, it will become deadlock.

Generally, I think a `<-pool.ctx.Done()` branch with `pool.readyConnection <- connection` will help to avoid blocking.  And `pool.ctx.Err() != nil` can replace `pool.closed`.

Reply:
Replacing pool.closed with ctx.Err is a good idea, I wiil update with it. Thanks.